test/tools/cluster.sh

```diff
@@ -0,0 +1,173 @@
+#!/bin/bash
+
+if [ "$#" -lt 1 ]; then
+    echo "usage: cluster <init|start|stop|prune|kill|uri|test> <single|replica|sharded>"
+    exit 0
+fi
+
+PORT_REPLICA=31000
+PORT_SHARDED=51000
+PORT_SINGLE=27017
+
+log () {
+  echo "$1 --------------------"
+}
+
+cluster () {
+    COMMAND="$1"
+    if [[ "$COMMAND" == "uri" ]]; then
+        TOPOLOGY=$2
+        if [[ "$TOPOLOGY" == "replica" ]] || [[ "$TOPOLOGY" == "replicaset" ]] || [[ "$TOPOLOGY" == "replica_set" ]]; then
+            MONGODB_URI="mongodb://localhost:$PORT_REPLICA/?replicaSet=rs"
+            export MONGODB_URI
+            echo $MONGODB_URI
+        elif [[ "$TOPOLOGY" == "sharded" ]] || [[ "$TOPOLOGY" == "sharded_clister" ]] || [[ "$TOPOLOGY" == "shardedcluster" ]] || [[ "$TOPOLOGY" == "mongos" ]]; then
+            ADD_PORT=$(( $PORT_SHARDED + 1 ))
+            MONGODB_URI="mongodb://localhost:$PORT_SHARDED,localhost:$ADD_PORT"
+            export MONGODB_URI
+            echo $MONGODB_URI
+        elif [[ "$TOPOLOGY" == "standalone" ]] || [[ "$TOPOLOGY" == "server" ]] || [[ "$TOPOLOGY" == "single" ]]; then
+            MONGODB_URI="mongodb://localhost:$PORT_SINGLE"
+            export MONGODB_URI
+            echo $MONGODB_URI
+        else
+            echo "uri subcommand needs at least one argument"
+            exit 1
+        fi
+    elif [[ "$COMMAND" == "init" ]]; then
+        TOPOLOGY=$2
+        if [[ "$TOPOLOGY" == "replica" ]] || [[ "$TOPOLOGY" == "replicaset" ]] || [[ "$TOPOLOGY" == "replica_set" ]]; then
+            log 'init replica'
+            mlaunch init --dir "./data/$PORT_REPLICA" --replicaset --nodes 3 --arbiter --name rs --port "$PORT_REPLICA" --enableMajorityReadConcern --setParameter enableTestCommands=1
+            cluster uri replica
+        elif [[ "$TOPOLOGY" == "sharded" ]] || [[ "$TOPOLOGY" == "sharded_clister" ]] || [[ "$TOPOLOGY" == "shardedcluster" ]] || [[ "$TOPOLOGY" == "mongos" ]]; then
+            log 'init sharded'
+            mlaunch init --dir "./data/$PORT_SHARDED" --replicaset --nodes 3 --arbiter --name rs --port "$PORT_SHARDED" --setParameter enableTestCommands=1 --sharded 1 --mongos 2
+            cluster uri sharded
+        elif [[ "$TOPOLOGY" == "standalone" ]] || [[ "$TOPOLOGY" == "server" ]] || [[ "$TOPOLOGY" == "single" ]]; then
+            log 'init single'
+            mlaunch init --dir "./data/$PORT_SINGLE" --single --setParameter enableTestCommands=1
+            cluster uri single
+        else
+            cluster init replica
+            cluster init sharded
+            cluster init single
+        fi
```

Comment:
I still think it can be useful 1) as the dependency to start and 2) understanding how `mlaunch` works. I'm down to hide it from the menu, but I don't think it's necessary.